plaso/parsers/amcache.py

```diff
@@ -1,331 +1,324 @@
 
     Args:
-      am_entry (pyregf.key): amcache File key.
       parser_mediator (ParserMediator): mediates interactions between parsers
           and other components, such as storage and dfvfs.
+      file_reference_key (pyregf.key): file reference key.
     """
-    amcache_datetime = am_entry.get_value_by_name(
-        self._AMCACHE_DATETIME).get_data_as_integer()
-    event_data = AmcacheEventData()
-
-    event_data.full_path = am_entry.get_value_by_name(
-        self._AMCACHE_FULL_PATH).get_data_as_string()
-    # Strip off the 4 leading zero's from the sha1 hash.
-    event_data.sha1 = am_entry.get_value_by_name(
-        self._AMCACHE_SHA1).get_data_as_string()[4:]
-
-    productname = am_entry.get_value_by_name(self._AMCACHE_PRODUCTNAME)
-    if productname:
-      event_data.productname = productname.get_data_as_string()
-
-    companyname = am_entry.get_value_by_name(self._AMCACHE_COMPANYNAME)
-    if companyname:
-      event_data.companyname = companyname.get_data_as_string()
-
-    fileversion = am_entry.get_value_by_name(self._AMCACHE_FILEVERSION)
-    if fileversion:
-      event_data.fileversion = fileversion.get_data_as_string()
-
-    languagecode = am_entry.get_value_by_name(self._AMCACHE_LANGUAGECODE)
-    if languagecode:
-      event_data.languagecode = languagecode.get_data_as_integer()
-
-    filesize = am_entry.get_value_by_name(self._AMCACHE_FILESIZE)
-    if filesize:
-      event_data.filesize = filesize.get_data_as_integer()
-
-    filedescription = am_entry.get_value_by_name(self._AMCACHE_FILEDESCRIPTION)
-    if filedescription:
-      event_data.filedescription = filedescription.get_data_as_string()
-
-    linkerts = am_entry.get_value_by_name(self._AMCACHE_LINKERTS)
-    if linkerts:
-      event_data.linkerts = linkerts.get_data_as_integer()
-
-    lastmodifiedts = am_entry.get_value_by_name(self._AMCACHE_LASTMODIFIEDTS)
-    if lastmodifiedts:
-      event_data.lastmodifiedts = lastmodifiedts.get_data_as_integer()
-
-    createdts = am_entry.get_value_by_name(self._AMCACHE_CREATEDTS)
-    if createdts:
-      event_data.createdts = createdts.get_data_as_integer()
-
-    programid = am_entry.get_value_by_name(self._AMCACHE_PROGRAMID)
-    if programid:
-      event_data.programid = programid.get_data_as_string()
-
-    event = time_events.DateTimeValuesEvent(
-        filetime.Filetime(amcache_datetime),
-        definitions.TIME_DESCRIPTION_MODIFICATION)
-    parser_mediator.ProduceEventWithEventData(event, event_data)
-
-    if event_data.createdts:
+    event_data = AMCacheFileEventData()
+
+    try:
+      if '0000' in file_reference_key.name:
+        # A NTFS file is a combination of MFT entry and sequence number.
+        sequence_number, mft_entry = file_reference_key.name.split('0000')
+        mft_entry = int(mft_entry, 16)
+        sequence_number = int(sequence_number, 16)
+        event_data.file_reference = '{0:d}-{1:d}'.format(
+            mft_entry, sequence_number)
+      else:
```

Comment:
What if it's not FAT? What if it's exFAT, or some other file system? I don't see that there's any checking that file_reference_key.name has to be a number, and the event data is a string, so please catch ValueError here and add it directly to the file reference.

Reply:
exfat and fat are the same family, other file systems like ReFS are currently unknown and will default to a single number.

ValueError is already caught at line 207 but I'll add TypeError